packages/interactor/src/interactor.ts

```diff
@@ -0,0 +1,86 @@
+import { converge } from './converge';
+import { InteractorSpecification } from './specification';
+import { Locator } from './locator';
+import { defaultOptions } from './options';
+import { NoSuchElementError, AmbigousElementError, NotAbsentError } from './errors';
+import { interaction, Interaction } from './interaction';
+
+export class Interactor<E extends Element> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private ancestors: Interactor<any>[] = [];
+
+  constructor(
+    public name: string,
+    private specification: InteractorSpecification<E>,
+    private locator: Locator<E>
+  ) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  find<T extends Interactor<any>>(interactor: T): T {
+    return Object.create(interactor, {
+      ancestors: {
+        value: [...this.ancestors, this, ...interactor.ancestors]
+      }
+    });
+  }
+
+  get description(): string {
+    return this.ancestors.slice().reverse().reduce((desc, interactor) => {
+      return `${desc} within ${interactor.name} ${interactor.locator.description}`
+    }, `${this.name} ${this.locator.description}`);
+  }
```

Comment:
Something to keep in mind for the future, but I can see a trace with more than two ancestors getting hard to read on one line. We may want to display this like how tsc shows type error traces.

Reply:
Totally agree! We should in general put more thought into the design of the errors than I've done here. If filters become a thing this will become even more unreadable.